modules/apps/commerce/commerce-shipment-web/src/main/java/com/liferay/commerce/shipment/web/internal/portlet/action/EditCommerceShipmentItemMVCActionCommand.java

```diff
@@ -280,13 +287,14 @@ else if ((commerceShipmentItem != null) &&
 				commerceShipmentItem =
 					_commerceShipmentItemService.updateCommerceShipmentItem(
 						commerceShipmentItem.getCommerceShipmentItemId(),
+						inventoryWarehouseExternalReferenceCode,
 						commerceInventoryWarehouseId, quantity, true);
 
 				if (BigDecimalUtil.eq(quantity, BigDecimal.ZERO)) {
 					commerceShipmentItem =
 						_commerceShipmentItemService.updateCommerceShipmentItem(
-							commerceShipmentItem.getCommerceShipmentItemId(), 0,
-							quantity, true);
+							commerceShipmentItem.getCommerceShipmentItemId(),
```

Comment:
my curiosity: what it does the update in this case?

Reply:
This is the case when quantity is different from the original one and it is also 0, so we have to update the ShipmentItem to have 0 quantity. Since quantity becomes 0, the warehouse stock must be restored (from where we took the item), so it reflects the original quantity before the shipment. This happens in _restoreStockQuantity() which is called from updateCommerceShipmentItem().

If I pass 0 as ID and null as ERC here, the warehouseId or warehouseERC should be taken from the commerceOrderItem which is retrieved from the db here:
CommerceShipmentItemLocalServiceImpl.updateCommerceShipmentItem()
...
CommerceOrderItem commerceOrderItem =
			_commerceOrderItemLocalService.getCommerceOrderItem(
				commerceShipmentItem.getCommerceOrderItemId());
...

If user would pass an incorrect warehouseId or warehouseERC here that would mean that we update (restore) the stock in the incorrect WH, thus I believe I should not pass these values here.